esp/logging/loggingagent/cassandraloggingagent/dbfieldmap.hpp

```diff
@@ -0,0 +1,90 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2014 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _DBFIELDMAP_HPP__
+#define _DBFIELDMAP_HPP__
+
+#pragma warning (disable : 4786)
+
+#include "jiface.hpp"
+#include "jstring.hpp"
+#include "jptree.hpp"
+
+class CLogField : public CInterface, implements IInterface
+{
+    StringAttr  name;
+    StringAttr  mapTo;
+    StringAttr  type;
+    StringAttr  defaultValue;
+
+public:
+    IMPLEMENT_IINTERFACE;
+    CLogField(const char* _name, const char* _mapTo, const char* _type)
+        : name(_name), mapTo(_mapTo), type(_type) {};
+    virtual ~CLogField() {};
+
+    const char* getName() { return name.get(); };
+    const char* getMapTo() { return mapTo.get(); };
+    const char* getType() { return type.get(); };
+    const char* getDefault() { return defaultValue.get(); };
+    void setDefault(const char* value) { defaultValue.set(value); };
+};
+
+class CLogTable : public CInterface
+{
+    StringAttr tableName;
+    CIArrayOf<CLogField> logFields;
+public:
+    IMPLEMENT_IINTERFACE;
+    CLogTable(const char* _tableName) : tableName(_tableName) {};
+    virtual ~CLogTable() {};
+
+    const char* getTableName() { return tableName.get(); };
+    void setTableName(const char* _tableName) { return tableName.set(_tableName); };
+    void loadMappings(IPropertyTree& cfg);
+    CIArrayOf<CLogField>& getLogFields() { return logFields; };
+};
+
+class CLogGroup : public CInterface, implements IInterface
+{
+    StringAttr name;
+    CIArrayOf<CLogTable> logTables;
+public:
+    IMPLEMENT_IINTERFACE;
+    CLogGroup(const char* _name) : name(_name) {};
+    virtual ~CLogGroup() {};
```

Comment:
what about the contents of the logTables, do the ClogTable elements need to be released?